openquake/hazardlib/gsim/drouet_Alpes_2015.py

```diff
@@ -0,0 +1,681 @@
+			   :class:`DrouetAlpes2015Rhypo_50bars`
+			   :class:`DrouetAlpes2015RjbHR_50bars`
+			   :class:`DrouetAlpes2015RrupHR_50bars`
+"""
+# 8 GMPEs for large magnitude stress parameter 100 bars (recommended by the authors)
+# including 4 for standard rock conditions vs30=800 m/s, kappa=0.03 s
+# and 4 GMPEs for hard rock conditions vs30=2000 m/s, kappa=0.01 s
+# The coefficients are published in the original paper and erratum
+# Additional 6 GMPEs for large magnitude stress parameter 50 bars
+# The coefficients are not published
+#
+from __future__ import division
+
+import numpy as np
+
+from openquake.hazardlib.gsim.base import CoeffsTable, GMPE
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+class DrouetAlpes2015Rjb(GMPE):
+    """
+    Implements GMPE developed by Douet & Cotton (2015) BSSA doi: 10.1785/0120140240.
+	This version is for a large magnitude stress parameters of 100 bars (recommended by the authors)
+    Valid for vs30=800 m/s
+    """
+
+    #: Supported tectonic region type is stable continental crust given that
+    #: the equations have been derived for Eastern North America.
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are spectral acceleration,
+    #: and peak ground acceleration, see table 6, page 1022 (PGA is assumed
+    #: to be equal to SA at 0.01 s)
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV,
+        SA
+    ])
+
+    #: Supported intensity measure component is the geometric mean of
+    #two : horizontal components
+    #:attr:`~openquake.hazardlib.const.IMC.AVERAGE_HORIZONTAL`,
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    #: Supported standard deviation type is only total, see equation 35, page
+    #: 1021
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT
+    ])
+
+    #: No site parameters are needed
+    REQUIRES_SITES_PARAMETERS = set()
+
+    #: Required rupture parameter is only magnitude, see equation 30 page
+    #: 1021.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', ))
+
+    #: Required distance measure is closest distance to rupture, see equation
+    #: 30 page 1021.
+    REQUIRES_DISTANCES = set(('rjb', ))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        assert all(stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+                   for stddev_type in stddev_types)
+
+        C = self.COEFFS[imt]
+        mean = self._compute_mean(C, rup.mag, dists.rjb)
+        if isinstance(imt, SA) or isinstance(imt, PGA): # Convert from m/s**2 to g
+            mean=mean-np.log(9.81)
```

Comment:
For converting between SI units and fractional g we prefer to use the scipy constants definition of g: `from scipy.constants import g`, to make sure we are not introducing small variations due to differences in rounding g

Reply:
Thanks Graeme, I did the modification.